be1-go/concurrent/waitgroup.go

```diff
@@ -0,0 +1,61 @@
+// Package concurrent defines data structures for keeping
+// track of open go-routines and returning from them cleanly
+// when the application is shut down.
```

Comment:
I would explain why you need that instead of a plain old `sync.WaitGroup`. The only additional feature I see is the ability to have a negative counter.

Reply:
It's to prevent a data race in the `/cli/witness/witness_test.go`. Calls to `server.ServeHTTP` handler and `server.Shutdown` are concurrent and the race detector produces a race on `wg.Add`, `wg.Wait`